testing/nand_sim/nand_sim_wrapper.c

```diff
@@ -0,0 +1,303 @@
+/****************************************************************************
+ * apps/testing/nand_sim/nand_sim_wrapper.c
```

Comment:
need move to nuttx/drivers/mtd, since it's a general log wrapper, not couple with nand_sim_raw directly, it's better to remove sim_ from the whole source file.

Reply:
Hi before this, there's [this logging-to-file issue](https://github.com/apache/nuttx/blob/9e22aafb0e39d37a868e5f0344eadb2f107bb4dd/Documentation/applications/testing/nand_sim/index.rst?plain=1#L178) that I need to sort out.

This seems to happen even when I flush the file in between. `stdout` works perfectly, but file doesn't.

I've even tried locks, thinking the issue might be due to the driver being killed before it could write the logs, but it doesn't seem to be case.

All of the [first 3 issues](https://github.com/apache/nuttx/blob/9e22aafb0e39d37a868e5f0344eadb2f107bb4dd/Documentation/applications/testing/nand_sim/index.rst?plain=1#L174-L181) stump me, and I can't find a way to get around it except maybe forks for the [third issue](https://github.com/apache/nuttx/blob/9e22aafb0e39d37a868e5f0344eadb2f107bb4dd/Documentation/applications/testing/nand_sim/index.rst?plain=1#L180).

Reply:
you can't call FILE* or fd related function in any kernel driver or module since FILE*/fd bind to a particular task/process context, but kernel driver/module can be called from different task/process, you should select one of following method:

1. Use file_open related kernel file function
2. Call syslog function and select file syslog channel

Reply:
Thanks a lot for this suggestion!